src/Interpreters/Access/InterpreterCreateRoleQuery.cpp

```diff
@@ -78,12 +93,27 @@ BlockIO InterpreterCreateRoleQuery::execute()
             new_roles.emplace_back(std::move(new_role));
         }
 
+        if (!query.storage_name.empty())
+        {
+            for (const auto & name : query.names)
+            {
+                auto id = access_control.find<Role>(name);
+
+                if (!id)
+                    continue;
+
+                auto another_storage_ptr = access_control.findStorage(*id);
```

Comment:
We can add a function in MultipleAccessStorage to contain this code with a name like ```MultipleAccessStorage::findExcludingStorage()`.